File: backend/entities/order.py
from typing import TypedDict, Any
from uuid import UUID

from entities.common import ParameterType

class OrderParameter(TypedDict):
    name: str
    type: ParameterType
    value: Any # depends on OrderParameterType(to serializing)

class Order(TypedDict):
    id: UUID
    payload: Any # serializable data
    parameters: list[OrderParameter]

File: frontend/src/models/classes/Order.ts
import type { Parametr } from "./Parametr";



export type AddProp = {
    additionalProp1: Object;
}

export class Order {
    id?: string;
    payload: AddProp;
    parameters: Parametr;
    constructor(payload: AddProp, parameters: Parametr) {
        this.payload = payload,
        this.parameters = parameters
    }
}